wp/plugin/wp.ml

```diff
@@ -190,8 +188,8 @@ let analyze_proj (ctx : Z3.context) (var_gen : Env.var_gen) (proj : project)
   let pre, env = Pre.visit_sub env post main_sub in
   let pre = Constr.mk_clause [Z3_utils.mk_smtlib2_single env flags.pre_cond] [pre] in
   let pre = Constr.mk_clause hyps [pre] in
-  (* Print statement for constraint-style prover output: *)
-  Format.printf "\nSub:\n%s\nPre:\n%!" (Sub.to_string main_sub);
+  if List.mem flags.show "bir" ~equal:String.equal then
```

Comment:
I assume you'll have something better than string equality once this is all wrapped up in a module?

Reply:
I'm thinking of something along the lines of:
```
type show =
  | Bir
  | RefutedGoals
  | Paths
  | PrecondInternal
  | PrecondSmtlib
```
that we can match on and possibly have it parsed into this type from the command line via http://binaryanalysisplatform.github.io/bap/api/odoc/bap/Bap/Std/Self/Config/index.html#val-converter